hazelcast-client/src/main/java/com/hazelcast/client/connection/nio/ClientConnection.java

```diff
@@ -60,6 +60,7 @@
     @Probe
     protected final int connectionId;
     private final AtomicBoolean alive = new AtomicBoolean(true);
+    private final AtomicBoolean closeCompeleted = new AtomicBoolean(false);
```

Comment:
Perhaps call it 'closed' ? Also the false initial value isn't needed, since it will be false be default.